scripts/GTF_OVERLAP_annotation.Rmd

```diff
@@ -0,0 +1,640 @@
+---
+title: "BEDTOOLS_INTERSECT_ANNOTATION_ANALYSIS"
+author: "Amarise Silié"
+date: "2024-03-07"
+output: html_document
+---
+
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(echo = TRUE)
+```
+
+# Load libraries
+
+```{r message=FALSE}
+# developed in R 4.3.2
+
+library(data.table) # v1.15.2, data storage
+library(stringi) # v1.8.3, for data manipulation
+library(ggplot2) # v3.5.0, for making figures
+library(tidyverse) # v2.0.0, data storage and manipulation, making figures
+library(scales) # v1.3.0, for making figures
+library(ggrepel) # v0.9.5 for making figures
+library(randomcoloR) # v1.1.0.1, for generation of palettes
+library(magrittr) #v2.0.3, for the forward-pipe operator
+```
+
+# Load data
+
+Loading the peak annotation files, which were created by intersecting the peak bed files with the GTF file from Ensembl. It is also possible to use annotation files created with the `scripts/get_gtfs.sh` and `scripts/get_features.sh` files.
+
+I loaded each file separately, but feel free to automate it if you have many files (I only had six). Make sure to set your work directory correctly to save the upcoming files in the right folders. You can also specify the path of your annotation data and the folder where you want to save your graphs below:
+
+```{r}
+data_path = '/path/to/data/' # <- do not forget the last /
+save_path = '/path/to/save/figures/'
+
+# this is how we will save our plots later:
+paste(save_path, 'name', sep='')
+
+# make directories beforehand, this may throw errors if R does not have the right permission
+dir.create(paste(save_path, 'basic_annotation_piechart', sep=''), showWarnings = FALSE)
+dir.create(paste(save_path, 'boxplots', sep=''), showWarnings = FALSE)
+dir.create(paste(save_path, 'genes', sep=''), showWarnings = FALSE)
+dir.create(paste(save_path, 'gene_types_barplots', sep=''), showWarnings = FALSE)
+dir.create(paste(save_path, 'heatmaps', sep=''), showWarnings = FALSE)
+dir.create(paste(save_path, 'histograms', sep=''), showWarnings = FALSE)
+dir.create(paste(save_path, 'scatterplots', sep=''), showWarnings = FALSE)
+```
+
+```{r}
+# loading the peaks intersecting with regions annotated with the "gene" feature
+pureclip_gof_3 = read.delim(paste(data_path, "gof_3_peaks_in_genes_ann.bed", sep=''))
+pureclip_gof_4 = read.delim(paste(data_path, "gof_4_peaks_in_genes_ann.bed", sep=''))
+pureclip_gof_5 = read.delim(paste(data_path, "gof_5_peaks_in_genes_ann.bed", sep=''))
+pureclip_wt_3 = read.delim(paste(data_path, "wt_3_peaks_in_genes_ann.bed", sep=''))
+pureclip_wt_4 = read.delim(paste(data_path, "wt_4_peaks_in_genes_ann.bed", sep=''))
+pureclip_wt_5 = read.delim(paste(data_path, "wt_5_peaks_in_genes_ann.bed", sep=''))
+
+# all tables to be combined later
+dfs = list(pureclip_gof_3 , pureclip_gof_4, pureclip_gof_5, pureclip_wt_3, pureclip_wt_4, 
+           pureclip_wt_5)
+names <- c("gof_1" ,"gof_2", "gof_3", "wt_1", "wt_2", "wt_3")
```

Comment:
In R it is generally preferred to use the `<-` operator for assignment (like in the last line), rather than `=`. I don't really mind of you prefer to use `=`, but it would be good to be consistent in this choice at least.